Upgraded software versions that are currently supported in Spack v0.19.0

ddeeptimahanti opened the pull request.

marcodelapierre commented:
@pelahi  when you see apparently non-sensical custom recipes, from the first deployment, I suggest you git clone spack, checkout v0.17.0, and do a diff vs the recipes there. It is very likely these are just additional versions/checksums.

pelahi commented:
@marcodelapierre that is my assumption too but I want to check with @ddeeptimahanti if that was the case. If would be good to add a README to the repo directory that will be ignored by spack but will allow us to better track why we have custom repos. Thoughts?

marcodelapierre commented:
I like your idea.
Mine was to have two distinct Pawsey recipe repos, one for checksums only, the other for proper recipe edits/fixes.
But I like your better because it is more concise, hence manageable.
Your readme would have a list/table of recipes, and for each one we would declare the recipe type: version checksum, bug fix, customisation, ..

dipietrantonio commented:
I think we just need to add comments on top of the recipe file describing our additions, easy.

dipietrantonio commented on repo/packages/awscli/package.py:
From the Slack conversation: 
```
Fixed awscli with newer versions of pyyaml and py-docutils
```

ddeeptimahanti replied:
Here are the changes @pelahi 

It's just updating versions of pyyaml and py-docutils that are compatible with python-3.10

diff --git a/var/spack/repos/builtin/packages/awscli/package.py b/var/spack/repos/builtin/packages/awscli/package.py
index 0378c94402..faa8b78f8b 100644
--- a/var/spack/repos/builtin/packages/awscli/package.py
+++ b/var/spack/repos/builtin/packages/awscli/package.py
@@ -17,14 +17,14 @@ class Awscli(PythonPackage):
     version("1.16.179", sha256="6a87114d1325358d000abe22b2103baae7b91f053ff245b9fde33cb0affb5e4f")
 
     depends_on("py-setuptools", type="build")
-    depends_on("py-botocore@1.13.44", when="@1.16.308", type=("build", "run"))
+    depends_on("py-botocore@1.21.12", when="@1.16.308", type=("build", "run"))
     depends_on("py-botocore@1.12.169", when="@1.16.179", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.15", type=("build", "run"))
+    depends_on("py-docutils@0.10:0.19", type=("build", "run"))
     depends_on("py-rsa@3.1.2:3.5.0", type=("build", "run"))
     depends_on("py-s3transfer@0.2.0:0.2", type=("build", "run"))
     depends_on("py-argparse@1.1:", when="^python@:2.6,3.0:3.1", type=("build", "run"))
     depends_on("py-pyyaml@3.10:3.13", when="^python@:2.6,3.0:3.3", type=("build", "run"))
-    depends_on("py-pyyaml@3.10:5.2", when="^python@3.4:", type=("build", "run"))
+    depends_on("py-pyyaml@3.10:6.0", when="^python@3.4:", type=("build", "run"))
     depends_on("py-colorama@0.2.5:0.3.9", when="^python@:2.6,3.0:3.3", type=("build", "run"))
     depends_on("py-colorama@0.2.5:0.4.1", when="^python@3.4:", type=("build", "run"))
     depends_on("py-nose", type="test")

ddeeptimahanti replied:
`diff --git a/var/spack/repos/builtin/packages/awscli/package.py b/var/spack/repos/builtin/packages/awscli/package.py
index 0378c94402..faa8b78f8b 100644
--- a/var/spack/repos/builtin/packages/awscli/package.py
+++ b/var/spack/repos/builtin/packages/awscli/package.py
@@ -17,14 +17,14 @@ class Awscli(PythonPackage):
     version("1.16.179", sha256="6a87114d1325358d000abe22b2103baae7b91f053ff245b9fde33cb0affb5e4f")
 
     depends_on("py-setuptools", type="build")
-    depends_on("py-botocore@1.13.44", when="@1.16.308", type=("build", "run"))
+    depends_on("py-botocore@1.21.12", when="@1.16.308", type=("build", "run"))
     depends_on("py-botocore@1.12.169", when="@1.16.179", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.15", type=("build", "run"))
+    depends_on("py-docutils@0.10:0.19", type=("build", "run"))
     depends_on("py-rsa@3.1.2:3.5.0", type=("build", "run"))
     depends_on("py-s3transfer@0.2.0:0.2", type=("build", "run"))
     depends_on("py-argparse@1.1:", when="^python@:2.6,3.0:3.1", type=("build", "run"))
     depends_on("py-pyyaml@3.10:3.13", when="^python@:2.6,3.0:3.3", type=("build", "run"))
-    depends_on("py-pyyaml@3.10:5.2", when="^python@3.4:", type=("build", "run"))
+    depends_on("py-pyyaml@3.10:6.0", when="^python@3.4:", type=("build", "run"))
     depends_on("py-colorama@0.2.5:0.3.9", when="^python@:2.6,3.0:3.3", type=("build", "run"))
     depends_on("py-colorama@0.2.5:0.4.1", when="^python@3.4:", type=("build", "run"))
     depends_on("py-nose", type="test")`

ddeeptimahanti replied:
In this, I updated versions of py-botocore, py-pyyaml and py-docutils

```
diff --git a/var/spack/repos/builtin/packages/awscli/package.py b/var/spack/repos/builtin/packages/awscli/package.py
index 0378c94402..faa8b78f8b 100644
--- a/var/spack/repos/builtin/packages/awscli/package.py
+++ b/var/spack/repos/builtin/packages/awscli/package.py
@@ -17,14 +17,14 @@ class Awscli(PythonPackage):
     version("1.16.179", sha256="6a87114d1325358d000abe22b2103baae7b91f053ff245b9fde33cb0affb5e4f")
 
     depends_on("py-setuptools", type="build")
-    depends_on("py-botocore@1.13.44", when="@1.16.308", type=("build", "run"))
+    depends_on("py-botocore@1.21.12", when="@1.16.308", type=("build", "run"))
     depends_on("py-botocore@1.12.169", when="@1.16.179", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.15", type=("build", "run"))
+    depends_on("py-docutils@0.10:0.19", type=("build", "run"))
     depends_on("py-rsa@3.1.2:3.5.0", type=("build", "run"))
     depends_on("py-s3transfer@0.2.0:0.2", type=("build", "run"))
     depends_on("py-argparse@1.1:", when="^python@:2.6,3.0:3.1", type=("build", "run"))
     depends_on("py-pyyaml@3.10:3.13", when="^python@:2.6,3.0:3.3", type=("build", "run"))
-    depends_on("py-pyyaml@3.10:5.2", when="^python@3.4:", type=("build", "run"))
+    depends_on("py-pyyaml@3.10:6.0", when="^python@3.4:", type=("build", "run"))
     depends_on("py-colorama@0.2.5:0.3.9", when="^python@:2.6,3.0:3.3", type=("build", "run"))
     depends_on("py-colorama@0.2.5:0.4.1", when="^python@3.4:", type=("build", "run"))
     depends_on("py-nose", type="test")
```

dipietrantonio commented:
Other than Pascal comments, looks good to me.

dipietrantonio reviewed (changes requested):
Add a comment at the top of each added/modified recipe stating why we have our own.

ddeeptimahanti commented on repo/packages/cp2k/package.py:
There is an issue with downloading 2022.1 
```
diff --git a/var/spack/repos/builtin/packages/cp2k/package.py b/var/spack/repos/builtin/packages/cp2k/package.py
index 34487891a9..f367d97cf4 100644
--- a/var/spack/repos/builtin/packages/cp2k/package.py
+++ b/var/spack/repos/builtin/packages/cp2k/package.py
@@ -17,7 +17,7 @@ class Cp2k(MakefilePackage, CudaPackage):
     """
 
     homepage = "https://www.cp2k.org"
-    url = "https://github.com/cp2k/cp2k/releases/download/v3.0.0/cp2k-3.0.tar.bz2"
+    url = "https://github.com/cp2k/cp2k/releases/download/v2022.1/cp2k-2022.1.tar.bz2"
     git = "https://github.com/cp2k/cp2k.git"
     list_url = "https://github.com/cp2k/cp2k/releases"
```

ddeeptimahanti commented on repo/packages/py-botocore/package.py:
updated py-docutils version

```
diff --git a/var/spack/repos/builtin/packages/py-botocore/package.py b/var/spack/repos/builtin/packages/py-botocore/package.py
index b754420f7b..eb0e61e681 100644
--- a/var/spack/repos/builtin/packages/py-botocore/package.py
+++ b/var/spack/repos/builtin/packages/py-botocore/package.py
@@ -26,7 +26,7 @@ class PyBotocore(PythonPackage):
 
     depends_on("py-setuptools", type="build")
     depends_on("py-jmespath@0.7.1:0", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.15", type=("build", "run"), when="@:1.17")
+    depends_on("py-docutils@0.10:0.19", type=("build", "run"), when="@:1.17")
     depends_on("py-ordereddict@1.1", type=("build", "run"), when="^python@2.6.0:2.6")
     depends_on("py-simplejson@3.3.0", type=("build", "run"), when="^python@2.6.0:2.6")
     depends_on("py-python-dateutil@2.1:2", type=("build", "run"))
```

ddeeptimahanti commented on repo/packages/py-docutils/package.py:
url change

```
diff --git a/var/spack/repos/builtin/packages/py-docutils/package.py b/var/spack/repos/builtin/packages/py-docutils/package.py
index 64f03ed834..f3431f69fa 100644
--- a/var/spack/repos/builtin/packages/py-docutils/package.py
+++ b/var/spack/repos/builtin/packages/py-docutils/package.py
@@ -15,8 +15,11 @@ class PyDocutils(PythonPackage):
     easy to read, easy to use, what-you-see-is-what-you-get plaintext
     markup language."""
 
-    homepage = "http://docutils.sourceforge.net/"
-    pypi = "docutils/docutils-0.15.2.tar.gz"
+#    homepage = "http://docutils.sourceforge.net/"
+#    pypi = "docutils/docutils-0.15.2.tar.gz"
+
+    homepage = "https://files.pythonhosted.org/packages/6b/5c/330ea8d383eb2ce973df34d1239b3b21e91cd8c865d21ff82902d952f91f/"
+    pypi = "docutils-0.19.tar.gz"
 
     version("0.19", sha256="33995a6753c30b7f577febfc2c50411fec6aac7f7ffeb7c4cfe5991072dcf9e6")
     version("0.18.1", sha256="679987caf361a7539d76e584cbeddc311e3aee937877c87346f31debc63e9d06")
```

dipietrantonio commented:
I noticed that we moved from singularity/3.8.6 to 3.8.7. Why don't we move to 3.10.2?

Also, the systems/setonix/settings.sh file needs to be updated to reflect these changes.

ddeeptimahanti commented:
> I noticed that we moved from singularity/3.8.6 to 3.8.7. Why don't we move to 3.10.2?
> 
> Also, the `systems/setonix/settings.sh` file needs to be updated to reflect these changes.

In this branch, I deleted only those custom recipes that are failing and modified only py-hatchet URL since it was failing. As discussed with you and @pelahi a couple of weeks ago, during the next iteration, we can only focus on assessing whether a custom recipe is still required or not.  

Our singularity recipe supports only until 3.8.7.

We have around ~50 custom recipes and it's too much work for this branch to see whether it's a version change or a bug fix. I would prefer to go with your idea of adding a comment at the start of the recipe so that it's easy to track. What do you think @marcodelapierre and @pelahi ?

dipietrantonio commented:
@ddeeptimahanti would you please update the `settings.sh` file to match the singularity version? it is still at 3.8.6.

I think adding comments on recipes is something we can start doing now and going forwards. In the rear future, we will need to revisit the existing recipes we have there to see what can be removed.
